File: hw/hidalgo/u3/hw35trategyPersistence/java/ToolsManagement/src/main/java/ec/edu/espe/tools/utils/Persistence.java
package ec.edu.espe.tools.utils;

import ec.edu.espe.tools.model.Tool;
import java.util.List;
/**
 *
 * @author Mikael Hidalgo, Paradigm, @ESPE
 */

public interface Persistence {
    public boolean create(Tool tool);
    public List<Tool> read(); 
    public boolean update(String id, Tool tool);
    public boolean delete(String id);
    public Tool find(String id);
}

File: hw/hidalgo/u3/hw35trategyPersistence/python/ec/edu/espe/tools_management/utils/persistence.py
from abc import ABC, abstractmethod
from typing import List
from model.tool import Tool

class Persistence(ABC):
    
    @abstractmethod
    def create(self, tool: Tool) -> bool:
        pass

    @abstractmethod
    def read(self) -> List[Tool]:
        pass

    @abstractmethod
    def update(self, tool: Tool) -> bool:
        pass

    @abstractmethod
    def delete(self, tool_id: str) -> bool:
        pass

    @abstractmethod
    def find(self, tool_id: str) -> Tool:
        pass